src/libraries/Microsoft.Extensions.Configuration.Binder/tests/SourceGenerationTests/Baselines/OptionsBuilder/Bind_T_BinderOptions.generated.txt

```diff
@@ -108,12 +108,19 @@ namespace Microsoft.Extensions.Configuration.Binder.SourceGeneration
         {
             ValidateConfigurationKeys(typeof(Program.MyClass), s_configKeys_ProgramMyClass, configuration, binderOptions);
 
-            instance.MyString = configuration["MyString"]!;
+            if (configuration["MyString"] is string value1)
+            {
+                instance.MyString = value1!;
+            }
 
             if (configuration["MyInt"] is string value2)
             {
                 instance.MyInt = ParseInt(value2, () => configuration.GetSection("MyInt").Path);
             }
+            else
+            {
+                instance.MyInt = default;
```

Comment:
Perhaps make this a conditional assignment to reduce LOC?

Reply:
A newer commit changed the `else` to `else if` so I don't think a conditional assignment works well anymore.